lib/ui/page/home/widget/scroll_keyboard_handler.dart

```diff
@@ -0,0 +1,277 @@
+      widget.scrollController.animateTo(
+        /// Safe clamp offset.
+        newOffset.clamp(0, widget.scrollController.position.maxScrollExtent),
+        duration: duration ?? _defaultLongScrollAnimationDuration,
+        curve: Curves.linear,
+      );
+
+  /// Scrolls up by [_scrollStep] amount.
+  void _scrollPageUp({bool quick = false}) {
+    final newOffset =
+        widget.scrollController.offset +
+        (widget.reverseList ? _scrollStep : -_scrollStep);
+    _animateTo(newOffset, quick ? _defaultQuickScrollAnimationDuration : null);
+  }
+
+  /// Scrolls down by [_scrollStep] amount.
+  void _scrollPageDown({bool quick = false}) {
+    final newOffset =
+        widget.scrollController.offset +
+        (widget.reverseList ? -_scrollStep : _scrollStep);
+    _animateTo(newOffset, quick ? _defaultQuickScrollAnimationDuration : null);
+  }
+
+  /// Handles repeated key events from the key handler.
+  void _handleRepeatedKey(LogicalKeyboardKey key) {
+    if (!widget.scrollController.hasClients) {
+      Log.debug(
+        'ScrollKeyboardHandler: ScrollController not attached to any scroll views',
+      );
+      return;
+    }
+
+    if (key == LogicalKeyboardKey.pageUp ||
+        (_isOptionPressed && key == LogicalKeyboardKey.arrowUp)) {
+      _scrollPageUp(quick: true);
+    } else if (key == LogicalKeyboardKey.pageDown ||
+        (_isOptionPressed && key == LogicalKeyboardKey.arrowDown)) {
+      _scrollPageDown(quick: true);
+    }
+  }
+
+  /// Handles all keyboard events from HardwareKeyboard.
+  bool _handleKeyEvent(KeyEvent event) {
+    if (!widget.scrollController.hasClients) {
+      Log.debug(
+        'ScrollKeyboardHandler: ScrollController not attached to any scroll views',
+      );
+      return false;
+    }
+
+    if (event is KeyDownEvent) {
+      final logicalKey = event.logicalKey;
+
+      // Track Option/Alt key state.
+      if (logicalKey == LogicalKeyboardKey.altLeft ||
+          logicalKey == LogicalKeyboardKey.altRight) {
+        _isOptionPressed = true;
+        return false;
+      }
+
+      /// Pass the event to the repeat click handler.
+      _keyHandler.onKeyEvent(event);
+
+      if (logicalKey == LogicalKeyboardKey.pageUp ||
+          (_isOptionPressed && logicalKey == LogicalKeyboardKey.arrowUp)) {
+        _scrollPageUp();
+        return true; // Event handled
+      } else if (logicalKey == LogicalKeyboardKey.pageDown ||
+          (_isOptionPressed && logicalKey == LogicalKeyboardKey.arrowDown)) {
+        _scrollPageDown();
+        return true; // Event handled
+      }
+    } else if (event is KeyUpEvent) {
+      final logicalKey = event.logicalKey;
+
+      /// Option (Alt) keyUp.
+      if (logicalKey == LogicalKeyboardKey.altLeft ||
+          logicalKey == LogicalKeyboardKey.altRight) {
+        _isOptionPressed = false;
+      }
+      _keyHandler.onKeyEvent(event);
+    } else if (event is KeyRepeatEvent) {
+      _keyHandler.onKeyEvent(event);
+    }
```

Comment:
Since `Flutter` already presents `KeyRepeatEvent`, which can be used to recreate the `Timer` for repeated key presses etc, then why is `_keyHandler` needed at all?